pretty-hydra.el

```diff
@@ -210,9 +221,55 @@ docstring."
                     (append heads `((,quit-key nil)))
                   heads))
          (body (pretty-hydra--remove-custom-opts body)))
-    `(defhydra ,name ,body
-       ,docstring
-       ,@heads)))
+    `(progn
+       (set (defvar ,(intern (format "%S/heads-plist" name))
+              nil
+              ,(format "heads-plist of %S." name))
+            (quote ,heads-plist))
+       (set (defvar ,(intern (format "%S/pretty-body" name))
+              nil
+              ,(format "pretty-body of %S." name))
+            (quote ,body))
+       (defhydra ,name ,body
+         ,docstring
+         ,@heads))))
+
+(defmacro pretty-hydra-define+ (name body heads-plist)
```

Comment:
Looks like most of the code is the same as `pretty-hydra-define` except for the `defhydra` vs `defhydra+` part and getting the existing `name/heads-plist` and `name/pretty-body`. Maybe extract a new function `pretty-hydra--generate` and call it from both `pretty-hydra-define` and `pretty-hydra-define+`?

Reply:
Good idea! I've pushed a another commit with this.

Reply:
Exactly what I had in mind. Thanks!